kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/AdapterGenerator.kt

```diff
@@ -541,26 +542,28 @@ internal class AdapterGenerator(
         MOSHI_UTIL, property.localName, property.jsonName, reader)
   }
 
-  private fun generateToJsonFun(): FunSpec {
+  private fun generateToJsonFun(ignoreToJson: Boolean): FunSpec {
     val result = FunSpec.builder("toJson")
         .addModifiers(KModifier.OVERRIDE)
         .addParameter(writerParam)
         .addParameter(valueParam)
-
-    result.beginControlFlow("if (%N == null)", valueParam)
-    result.addStatement("throw·%T(%S)", NullPointerException::class,
-        "${valueParam.name} was null! Wrap in .nullSafe() to write nullable values.")
-    result.endControlFlow()
-
-    result.addStatement("%N.beginObject()", writerParam)
-    nonTransientProperties.forEach { property ->
-      // We manually put in quotes because we know the jsonName is already escaped
-      result.addStatement("%N.name(%S)", writerParam, property.jsonName)
-      result.addStatement("%N.toJson(%N, %N.%N)",
-          nameAllocator[property.delegateKey], writerParam, valueParam, property.name)
+    if (ignoreToJson) {
+      result.addComment("ignoreToJson specified!")
```

Comment:
This should throw

Reply:
Absolutely (as said in description) but I found ```JsonDataException``` description very precise and not matching this case, and did not know what to throw so went with comment first to see real life gains.
Will update PR with JsonDataException as in all cases the gains are too big for ignoring them and worth maintaining a fork if necessary. (Let me know if you have better idea for what to throw)

Reply:
I would think `UnsupportedOperationException` is probably most appropriate here